dashboard/alerts/alert.py

```diff
@@ -41,19 +68,24 @@ def __init__(self, pkt: packet, alert_description="", alert_type=ALERT_TYPE.UNKN
         else:
             if "IP" in pkt:
                 self.device_ip = pkt["IP"].src
+            else:
+                self.device_ip = "[Layer 2]"
             if "Ethernet" in pkt:
                 self.device_mac = pkt["Ethernet"].src
+            else:
+                self.device_mac = "[Unknown]"
 
         # Use some magic config trickery to get the name for this device, otherwise unknown
         if "magic":
             self.device_name = "Unknown"
 
         self.description = alert_description
-        self.id = hash(self.device_mac + str(self.timestamp))
+        hasher = hashlib.sha1()
+        hasher.update(str(self.device_mac + self.timestamp).encode('utf-8'))
+        self.id = int(hasher.hexdigest()[:4], 16)
 
         # If there is raw information, try to save it
-        if "Raw" in pkt:
-            self.payload_info = pkt["Raw"].load
+        self.payload_info = hexdump(pkt, dump=True)
```

Comment:
Can you explain this change? Since in this constructor the packet is present I believe scapy will always have the 'raw' field for the payload.

Reply:
The "Raw" field in SCAPY is not used how you would intuitively think. Instead of being a raw representation of the packet, it is only the raw contents of TCP/UDP packets. So any flagged packets not using one of those protocols would not have this field

Reply:
That makes sense, thanks for explaining!